components/event-sources/reconciler/httpsource/reconcile.go

```diff
@@ -279,27 +326,31 @@ func (r *Reconciler) getOrCreatePolicy(src *sourcesv1alpha1.HTTPSource,
 	return policy, nil
 }
 
-// makeKnService returns the desired Knative Service object for a given
-// HTTPSource. An optional Knative Service can be passed as parameter, in which
+// makeDeployment returns the desired Deployment object for a given
+// HTTPSource. An optional Deployment can be passed as parameter, in which
 // case some of its attributes are used to generate the desired state.
-func (r *Reconciler) makeKnService(src *sourcesv1alpha1.HTTPSource,
-	sinkURI, loggingCfg, metricsCfg string) *servingv1alpha1.Service {
+func (r *Reconciler) makeDeployment(src *sourcesv1alpha1.HTTPSource,
+	sinkURI, loggingCfg, metricsCfg string) *appsv1.Deployment {
 
-	return object.NewService(src.Namespace, src.Name,
+	return object.NewDeployment(src.Namespace, src.Name,
```

Comment:
does it makes sense to add a finaliser to the created resource?

Reply:
Can you explain why it makes sense to have a finalizer on the deployment? In case the deployment get's deleted it is reconciled and recreated.